src/edu/washington/escience/myria/column/builder/BlobColumnBuilder.java

```diff
@@ -0,0 +1,190 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import com.almworks.sqlite4java.SQLiteException;
+import com.almworks.sqlite4java.SQLiteStatement;
+import com.google.common.base.Preconditions;
+
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.BlobColumn;
+import edu.washington.escience.myria.column.mutable.BlobMutableColumn;
+import edu.washington.escience.myria.proto.DataProto.BlobColumnMessage;
+import edu.washington.escience.myria.proto.DataProto.ColumnMessage;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleUtils;
+import edu.washington.escience.myria.util.MyriaUtils;
+/**
+ * A column of Blob values.
+ *
+ */
+public final class BlobColumnBuilder extends ColumnBuilder<ByteBuffer> {
+
+  /**
+   * The internal representation of the data.
+   * */
+  private final ByteBuffer[] data;
+
+  /** Number of elements in this column. */
+  private int numBB;
+
+  /**
+   * If the builder has built the column.
+   * */
+  private boolean built = false;
+
+  /** Constructs an empty column that can hold up to TupleBatch.BATCH_SIZE elements. */
+  public BlobColumnBuilder() {
+    numBB = 0;
+    data = new ByteBuffer[TupleUtils.getBatchSize(Type.BLOB_TYPE)];
+  }
+
+  /**
+   * copy.
+   *
+   * @param numDates the actual num strings in the data
+   * @param data the underlying data
+   * */
+  private BlobColumnBuilder(final ByteBuffer[] data, final int numBB) {
+    this.numBB = numBB;
+    this.data = data;
+  }
+
+  /*
+   * Constructs a DateColumn by deserializing the given ColumnMessage.
+   *
+   * @param message a ColumnMessage containing the contents of this column.
+   *
+   * @param numTuples num tuples in the column message
+   *
+   * @return the built column
+   */
+  public static BlobColumn buildFromProtobuf(final ColumnMessage message, final int numTuples) {
+    Preconditions.checkArgument(
+        message.getType().ordinal() == ColumnMessage.Type.BLOB_VALUE,
+        "Trying to construct Bytes from non-bytes ColumnMessage %s",
+        message.getType());
+
+    Preconditions.checkArgument(message.hasBlobColumn(), "ColumnMessage is missing DateColumn");
+    final BlobColumnMessage BlobColumn = message.getBlobColumn();
+    ByteBuffer[] newData = new ByteBuffer[numTuples];
+    ByteBuffer data = BlobColumn.getData().asReadOnlyByteBuffer();
+    for (int i = 0; i < numTuples; i++) {
+      newData[i] = ByteBuffer.allocate(data.capacity());
+      newData[i].put(data);
+    }
+
+    return new BlobColumnBuilder(newData, numTuples).build();
+  }
+
+  @Override
+  public BlobColumnBuilder appendBlob(final ByteBuffer value) throws BufferOverflowException {
+    Preconditions.checkState(
+        !built, "No further changes are allowed after the builder has built the column.");
+    Objects.requireNonNull(value, "value");
+    if (numBB >= TupleUtils.getBatchSize(Type.BLOB_TYPE)) {
```

Comment:
It's probably unnecessary since you will get an `IndexOutOfBound` later in `data[numBB+]`.